conans/model/options.py

```diff
@@ -139,6 +139,8 @@ def rm_safe(self, field):
             delattr(self, field)
         except ConanException:
             pass
+        # except KeyError:
```

Comment:
It would make sense to open a separate PR with the simplest integration test that can reproduce the potential need for this.

Reply:
After checking the issue that led to this change, it seems that it is not working properly with the "virtual/consumert" conanfile. In that case, the attribute "_constrained" seems to be false and the `ConanException` is not raising at `_ensure_exists()`.

I don't know how this `_constrained` attribute works, so I don't know how to proceed with that.

-----

Details of the debugging:

1. "Virtual/Consumer" conanfile does not have `config_options()` method defined, so it executes the `handle_common_config_options() method` here: https://github.com/conan-io/conan/blob/07cfcc37c62abc04ab80a0e72f755402d3b9710d/conans/client/conanfile/configure.py#L11 (not this line but the ``else`` clause proposed in this PR).

2. Then, `self.options.rm_safe("fPIC")` is executed by the helper: https://github.com/conan-io/conan/blob/07cfcc37c62abc04ab80a0e72f755402d3b9710d/conans/model/options.py#L137

3. Then, the `ensure_exists()` method is called https://github.com/conan-io/conan/blob/07cfcc37c62abc04ab80a0e72f755402d3b9710d/conans/model/options.py#L175

4. As the `_constrained` attribute is False at this point https://github.com/conan-io/conan/blob/07cfcc37c62abc04ab80a0e72f755402d3b9710d/conans/model/options.py#L159, the ConanException is not raised and not captured by `rm_safe()`. Instead, a `KeyError` is thrown when actually deleting the non-existing option at https://github.com/conan-io/conan/blob/07cfcc37c62abc04ab80a0e72f755402d3b9710d/conans/model/options.py#L176

------

I have opened a small PR with the isolated change of the keyerror to see if it affects the current behavior https://github.com/conan-io/conan/pull/14226

Also, if this is not the right approach I will need some help to understand how `_constrained` attribute works